lib/puppet/provider/mysql.rb

```diff
@@ -4,7 +4,7 @@ class Puppet::Provider::Mysql < Puppet::Provider
   initvars
 
   # Make sure we find mysqld on CentOS
-  ENV['PATH']=ENV['PATH'] + ':/usr/libexec'
+  ENV['PATH']=ENV['PATH'] + ':/usr/libexec:/usr/sbin'
```

Comment:
Because it was already in your path. In some minimalist environment it can be absent.